erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -393,6 +395,18 @@ def check_credit_limit(self):
 		if validate_against_credit_limit:
 			check_credit_limit(self.customer, self.company, bypass_credit_limit_check_at_sales_order)
 
+	def unlink_timesheets(self):
+		for row in self.timesheets:
+			timesheet = frappe.get_doc('Timesheet', row.time_sheet)
+			for time_log in timesheet.time_logs:
+				if time_log.sales_invoice == self.name:
+					time_log.sales_invoice = None
```

Comment:
See if you can directly unlink using an UPDATE query. Check reference [here](https://github.com/frappe/erpnext/blob/14ca9ce1b7e0b3389b3c759200e0febc86be9025/erpnext/loan_management/doctype/loan/loan.py#L142-L148).

Reply:
We can't do that. Some fields need to recalculate billed hours,  %billed, and status to update.